File: frontend/src/core/apiError.ts
export interface ApiErrorBody {
  code: string
  messageKey: string
  parameters: Record<string, string | number>
  details: string | null
}

export function isApiErrorBody(value: unknown): value is ApiErrorBody {
  if (typeof value !== 'object' || value === null) {
    return false
  }
  const candidate = value as Record<string, unknown>
  return typeof candidate.code === 'string' && typeof candidate.messageKey === 'string'
}

File: backend/GastronomyApp.Api/ErrorHandling/ApiError.cs
namespace GastronomyApp.Api.ErrorHandling;

public sealed record ApiError
{
  public required string Code { get; init; }

  public required string MessageKey { get; init; }

  public required IReadOnlyDictionary<string, string> Parameters { get; init; }

  public string? Details { get; init; }
}

public sealed record ProblemDescription
{
  public required int StatusCode { get; init; }

  public required ApiError Error { get; init; }
}
